src/inmem/mutable.rs

```diff
@@ -31,6 +31,19 @@ pub(crate) type MutableScan<'scan, R> = Range<
     Option<R>,
 >;
 
+/// Signals to memtable whether size threshold has been reached and needs compaction
+#[derive(Debug)]
+pub enum WriteResult {
+    Continue,
+    NeedCompaction,
+}
+
+impl WriteResult {
+    pub fn needs_compaction(&self) -> bool {
```

Comment:
adding this mainly to make python binding build success, but end up using this in other parts too.